### DISCUSSION TITLE
Will there be any support for multi-tenancy in the near future?

### DISCUSSION THREAD

**Discussion by sasokovacic** [lucidarch/lucid#10] [Q&A]
Hi, I'm wondering if you are planning to add support for multi-tenancy... I'm building multi-tenancy apps and would really like to build them in a "lucid" way.. I use Laravel Tenancy package..
The main challenge I'm facing  is the file structure... in a multi-tenancy apps it is recommended to separate the tenant app and system app  (that is used for managing tenants) but as I can see the lucid console is building it's own file structure... Can you give me any advice on how to achieve this in case you are not planning on adding support for multi-tenancy apps? How can I change the location of generated classes?

**Comment by Mulkave** [comment]
Interesting!

Currently i would say the best approach would be to use a Lucid Monolith to build your application, which may help you achieve a similar result with multi-tenancy, especially if you prefer to segregate the application's facets through services.

See here for more on conditionally registering services: https://docs.lucidarch.dev/services/#register-service
